spec/unit/cli_spec.rb

```diff
@@ -75,6 +75,44 @@
 
       run_command(["inspect-container", "docker_image_foo"])
     end
+
+    it "calls .check_container_name and raises when the image and name consist slashes" do
+      expect(Cli).to receive(:check_container_name!).with("docker/foo", "docker/foo")
+      # unlike the real world raise doesn't stop the program flow in this CLI tests and
+      # it is not possible to check for a raise with `run_command`
+      # expect_any_instance_of(InspectTask).not_to receive(:inspect_system)
+
+      run_command(["inspect-container", "docker/foo"])
+    end
+
+    it "calls .check_container and continues if only the image consists slashes" do
+      expect(Cli).to receive(:check_container_name!).with("docker/foo", "docker_foo")
+      expect_any_instance_of(InspectTask).to receive(:inspect_system).
+        with(
```

Comment:
Since this test is only about verifying that `inspect_system` is called I'd omit the expectations on the arguments.